### ISSUE TITLE
Request for Ros 2 Support

### ISSUE THREAD

**Issue by anthonytec2** [Kinovarobotics/ros_kortex#69]
ROS 2 support for all the attached packages

**Comment by alexvannobel** [comment]
Hi @anthonytec2 ,

I don't have a date to give you but we consider adding ROS2 support. I'll keep this issue updated as I have more info to give you.

Thanks for the feedback,
Alex

**Comment by ngeerlingsriwo** [comment]
Hi @alexvannobel,

Are there any updates regarding ROS2 support?

**Comment by felixmaisonneuve** [comment]
Hi @ngeerlingsriwo,

There is still no change regarding ROS2. 
We are prioritizing moving to ROS Neotic first (we are currently using ROS Melodic), but we still consider adding ROS2 support.

Best,
Felix

**Comment by civerachb-cpr** [comment]
Since it appears there's a Noetic branch available now, is there an updated timeline on when we might see ROS2 support?  Or a definitive "we've considered it, but have decided against it" answer?

**Comment by alexvannobel** [comment]
Hi @civerachb-cpr ,

We have not decided against developing a ROS2 driver. However, before doing so and blindly adapting our ROS1 driver to ROS2 (which is a considerable effort already), we have to assess a couple elements such as (but not limited to):
- Defining requirements based on users needs;
- Assessing the workload including testing strategy;
- Allocating the necessary bandwidth to accomplish the work

In short : I don't have a timeline to give you, but I can assure you we're looking into it to understand how we should develop such a driver and we also understand the transition from ROS1 to ROS2 for our users and the robotics world at large will come sooner than later.

Best regards,
Alex

**Comment by civerachb-cpr** [comment]
Thanks for the clarification!  I'm glad a ROS2 driver is still on the table, even if there isn't a timeline in place yet.

**Comment by civerachb-cpr** [comment]
It's almost a year later. Are we any closer to a timeline for ROS2 support?

**Comment by felixmaisonneuve** [comment]
Hi @civerachb-cpr,
We can confirm a ROS2 implementation will be done for Gen3, but do not have a release timeline yet.

I will notify that clients are asking for this and I can recommand you to contact your Kinova sales representative (since you are from Clearpath Robotics) and ask for this too. 
This is not my decision, but with pressure things might start moving faster. 

Regards,
Felix

**Comment by ngeerlingsriwo** [comment]
Will Ubuntu 22.04/ROS2 Humble LTS be supported in the ROS2 implementation?

**Comment by felixmaisonneuve** [comment]
ROS2 support will be for Ubuntu22.04

**Comment by felixmaisonneuve** [comment]
ROS2 support is now available for Ubuntu22.04.
You can find it here : https://github.com/Kinovarobotics/ros2_kortex

Closing this issue
